api/android/api/src/main/java/org/nnsuite/nnstreamer/SingleShot.java

```diff
@@ -168,6 +193,42 @@ public void setTimeout(int timeout) {
         }
     }
 
+    /**
+     * Sets the information (tensor dimension, type, name and so on) of input data for the given model.
+     * Updates the output information for the model internally.
+     *
+     * @param in The input tensors information
+     *
+     * @throws IllegalStateException if failed to set the input information
+     * @throws IllegalArgumentException if given param is null
+     */
+    public void setInputInfo(@NonNull TensorsData in) {
+        checkPipelineHandle();
+
+        if (in == null) {
+            throw new IllegalArgumentException("Input tensor data is null");
+        }
+
+        if (!nativeSetInputInfo(mHandle, in)) {
+            throw new IllegalStateException("Failed to set input tensor info");
+        }
+    }
+
+    /**
+     * Updates the information (tensor dimension, type, name and so on) of input data, and get the updated output information for the given model.
+     *
+     * @param in The input tensors information
+     *
+     * @return The output tensors information
+     *
+     * @throws IllegalStateException if failed to set the input information
+     * @throws IllegalArgumentException if given param is null
+     */
+    public TensorsInfo updateInputInfo(@NonNull TensorsData in) {
```

Comment:
Anyway, is this necessary? IMO, updateInputInfo and setInputInfo is duplicated.
How about adding only one method to change input dimension?
I think that we don't need additional param in invoke() method.

```
Single single = new SingleShot(modelfile);

single.setInputInfo(newInfo);
single.invoke(inData);

outInfo = single.getOutputInfo(); <- this will return changed output meta.

```

Reply:
`UpdateInputInfo` removed. I just added it to have same equivalent APIs as in C.

Regarding additional param in invoke() method, they are just added to make API more flexible (both the additional added parameters are `Nullable`) and to match equivalent C API functions as given in #1806.

Reply:
Then, how about adding just one method - setInputInfo()?
Developer can set (or update) dimension with following steps:
```
1. open with changed dimension 
SingleShot single = new SingleShot (model, in, out); <-- input and output dimension can be changed during this method call

single.invoke (data);

2. open with default dimension
SingleShot single = new SingleShot(model);

single.setInputInfo (inInfo); <-- input and output dim will be changed

single.invoke (data);
```